sdk/eventhub/azure-eventhub/azure/eventhub/_common.py

```diff
@@ -207,12 +221,30 @@ def properties(self, value):
 
     @property
     def system_properties(self):
-        # type: () -> Dict[Union[str, bytes], Any]
+        # type: () -> Dict[bytes, Any]
         """Metadata set by the Event Hubs Service associated with the event
 
         :rtype: dict
         """
-        return self.message.annotations
+
+        if not self._sys_properties:
+            self._sys_properties = {
+                PROP_MESSAGE_ID: self.message.properties.message_id,
+                PROP_USER_ID: self.message.properties.user_id,
+                PROP_TO: self.message.properties.to,
+                PROP_SUBJECT: self.message.properties.subject,
+                PROP_REPLY_TO: self.message.properties.reply_to,
+                PROP_CORRELATION_ID: self.message.properties.correlation_id,
+                PROP_CONTENT_TYPE: self.message.properties.content_type,
+                PROP_CONTENT_ENCODING: self.message.properties.content_encoding,
+                PROP_ABSOLUTE_EXPIRY_TIME: self.message.properties.absolute_expiry_time,
+                PROP_CREATION_TIME: self.message.properties.creation_time,
+                PROP_GROUP_ID: self.message.properties.group_id,
+                PROP_GROUP_SEQUENCE: self.message.properties.group_sequence,
+                PROP_REPLY_TO_GROUP_ID: self.message.properties.reply_to_group_id,
+            }
+            self._sys_properties.update(self.message.annotations)
```

Comment:
why we're update the annotation here -- other languages are all having these information on "message.annotation"?

Reply:
I forget whether we support sending a received event out, but if we support that, then what will "message.properties" or "message.annotations" looks like if we receive that event again.

Reply:
oops, my bad, `a.update(b)` means `a` being updated.